bin/codemods/src/assign-component-to-displayname-variable.js

```diff
@@ -0,0 +1,160 @@
+/*
+ This codemod updates
+
+ export default someHoC( React.createClass( {
+    displayName: 'SomeDisplayName',
+ } ) );
+
+ to
+
+ const SomeDisplayName = React.createClass({
+   displayName: 'SomeDisplayName',
+ });
+
+ export default someHoC( SomeDisplayName );
+
+ and
+
+ export default someHoC( class extends {Component|PureComponent|React.Component} {
+    static displayName = 'SomeDisplayName';
+ } );
+
+ to
+
+ class SomeDisplayName extends {Component|PureComponent|React.Component} {
+    static displayName = 'SomeDisplayName';
+ };
+
+ export default someHoC( SomeDisplayName );
+ */
+
+const createClassIdentifier = {
+    type: 'CallExpression',
+    callee: {
+        type: 'MemberExpression',
+        object: {
+            name: 'React',
+        },
+        property: {
+            name: 'createClass',
+        }
+    },
+};
+
+const displayNamePropertyIdentifier = {
+    key: {
+        name: 'displayName',
+    },
+};
+
+const isValidComponentString = string =>
+    string === 'Component' ||
+    string === 'PureComponent';
+
+const argIsCreateClassInstance = arg =>
+        arg.get( 'callee' , 'object', 'name' ).value === 'React' &&
+        arg.get( 'callee' , 'property', 'name' ).value === 'createClass';
+
+const argIsExtendsComponentInstance = arg =>
+    arg.get( 'type' ).value === 'ClassExpression' && (
+        isValidComponentString( arg.get( 'superClass', 'name' ).value ) ||
+        arg.get( 'superClass', 'object', 'name' ).value === 'React' &&
+        isValidComponentString( arg.get( 'superClass', 'property', 'name' ).value )
+    );
+
+const argIsComponent = arg =>
+    argIsCreateClassInstance( arg ) ||
+    argIsExtendsComponentInstance( arg );
+
+const getMatchingArg = args =>
+    args.filter( argIsComponent )[ 0 ];
+
+export default function transformer( file, api ) {
+    const j = api.jscodeshift;
+    const root = j( file.source );
+
+    // string => object =>
+    const replaceClassOrGetValue = displayName => arg =>
+        argIsComponent( arg )
+            ? j.identifier( displayName )
+            : arg.value;
+
+    const exportDefaultDeclarations = root
```

Comment:
This should always return either 0 or 1 result.  We could return early in the case of 0 and not have to write the rest in terms of a loop with:

```js
const defaultExportDeclaration = _.head(
  root.find( j.ExportDefaultDeclaration )
);

if ( ! defaultExportDeclaration ) {
  root.toSource(); // not sure if this line is correct
}

// continue rest without for loop
const calleeNameValue = node.get( 'declaration', 'callee', 'name' ).value;
...
```

Reply:
I do kind of agree, though a .forEach over a 0 length array would surely be inconsequential, nothing would be run right?
I did originally have this early return in place but figured it'd actually be safer to run in a forEach.
Happy to refactor though :)